### PR TITLE
Adding Blockly blocks to Content levels.

### PR DESCRIPTION
**Pull request by daynew**
# Description 
In Content levels, such as External levels, level creators were taking
screenshots of Blockly blocks and adding them to the instructions in
order to give a preview to the student of what the blocks look like.
This has become a problem because screenshots are not translatable and
we are expanding our content to be used internationally.
* Added common and turtle CDO blocks to Content levels.
* <xml> blocks in Markdown content render as Blockly blocks.

### Example of translations working
image
### Working examples of all the existing screenshots used in External levels.
image

## Links
- [jira](https://codedotorg.atlassian.net/browse/FND-414)

## Testing story
* Added UI/eyes test

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/sites/studio/pages/levels/_content.js [hunk 1]
```diff
@@ -1,19 +1,47 @@
...
+  // Content levels don't have a particular 'app' associated with them so make
+  // available all the different app specific Blockly blocks as they are needed
+  // by level creators. This will enable them to write Blockly XML and render
+  // the blocks directly in the level text.
+  Blockly.assetUrl = assetUrl;
+  Blockly.Css.inject(document);
+  // Install the common Blockly blocks
+  commonBlocks.install(window.Blockly, {});
+  // Install the custom CDO blocks for the following level/app types.
+  var options = [{skin: {id: 'birds'}, app: 'maze'}, {skin: {}, app: 'turtle'}];
```
Comment (Hamms) [inline apps/src/sites/studio/pages/levels/_content.js]: note this is inconsistent with the above comment, which claims that we are making _all_ the apps available.

I don't think I'm in favor of this approach; in particular, the existing app-specific locale files are not intended to be used together and there is no guarantee that they won't have conflicts. I'd recommend that we install only the common blockly blocks for now, and as follow-up work look into either adding an app association to content levels or unifying the app locales in general.

File: apps/src/sites/studio/pages/levels/_content.js [hunk 2]
```diff
@@ -1,19 +1,47 @@
...
 
+    var container = this;
     ReactDom.render(
       React.createElement(SafeMarkdown, this.dataset, null),
-      this
+      this,
+      function() {
+        // After the Markdown is rendered, render any Blockly blocks defined in
+        // <xml> blocks.
+        convertXmlToBlockly(container);
```
Comment (Hamms) [inline apps/src/sites/studio/pages/levels/_content.js:38]: nit; I'd recommend either

```js
    const container = this;
    ReactDom.render(
      React.createElement(SafeMarkdown, container.dataset, null),
      container,
      function() {
        // After the Markdown is rendered, render any Blockly blocks defined in
        // <xml> blocks.
        convertXmlToBlockly(container);
      }
    }
```

or

```js
    ReactDom.render(
      React.createElement(SafeMarkdown, this.dataset, null),
      this,
      function() {
        // After the Markdown is rendered, render any Blockly blocks defined in
        // <xml> blocks.
        convertXmlToBlockly(this);
      }
    }
```

for consistency

File: TESTING.md [hunk 3]
```diff
@@ -106,7 +106,7 @@ If you get an error about missing db fields, try migrating your test database:
 We have a set of integration tests, divided into "UI tests" (Selenium+Cucumber) and "Eyes tests" (Selenium+Cucumber+Applitools).  These tests live in [dashboard/test/ui](dashboard/test/ui) - for information on setting up and running these tests, see [the README in that directory](dashboard/test/ui) and our [guide to adding an eyes test](docs/testing-with-applitools-eyes.md).
 Or you can just use this shortcut (after you've installed chromedriver):
 
-`rake test:ui feature=dashboard/test/ui/features/sometest.feature`
+`bundle exec rake test:ui feature=dashboard/test/ui/features/sometest.feature`
```
Comment (Hamms) [inline TESTING.md:109]: :+1:

File: apps/src/sites/studio/pages/levels/_content.js (hunk 1 above)
Comment (Hamms) [reply apps/src/sites/studio/pages/levels/_content.js]: s/locale/block

Comment (daynew) [reply apps/src/sites/studio/pages/levels/_content.js]: From our conversation offline, I have created TODO's and a follow up JIRA task which we will get prioritized for Jan 2020.
https://codedotorg.atlassian.net/browse/FND-972

Comment (daynew) [reply apps/src/sites/studio/pages/levels/_content.js]: I talked with Bryan and found out he is mostly interested in being able to render Turtle Blockly blocks. So I have removed the second level type "maze" from the blocks imported. This way we don't have to do cleanup in January and we will still do the dynamic level type loading later in H1.

**Review by Hamms — APPROVED** [review]
Looks great! Thanks for following up on the 'multiple block types' thing, I'm much happier with where we landed.

**Comment by bcjordan** [comment]
This is so cool!! Remember always wanted to support this 😄

**Pull request merged**